### PR TITLE
Check that Oxpoints returns a location for a given oxpoints ID

### PR DESCRIPTION
**Pull request by birniefoxford**
If an event has a location which references a deleted oxpoint, it can't return a location name when generating a calendar. Added a check to ensure that the location references a valid oxpoint before proceeding, otherwise returning "Venue to be announced".

### CODE REVIEW

File: talks/api/serializers.py [hunk 1]
```diff
@@ -59,14 +59,15 @@ def get_full_url(self, obj):
...
         if event.location:
-            location = event.api_location
-            name = location['name']
-            if event.location_details:
-                name += " (" + event.location_details + ")"
-            location_string = name
-            if location.get('address'):
-                location_string += ", " + location.get('address')
-            return location_string
+            if event.api_location:
```
Comment (ahaith) [inline talks/api/serializers.py]: Is there anything else to fall back on if this isn't here? Perhaps location_details, as in the case where there is no location?

**Comment by birniefoxford** [comment]
Changed elif to if on line 71 of talks/api/serializers.py. This way, if api_location doesn't work it will try to fall back to location_details, and failing that use "Venue to be announced".

**Comment by ahaith** [comment]
Thanks Finlay!

**Comment by envycontent** [comment]
Fantastic - thanks!

Anne

Anne Bowtell | Web Integration Manager
Medical Sciences Division | University of Oxford | Level 3, John Radcliffe Hospital, Oxford, OX3 9DU
anne.bowtell@medsci.ox.ac.uk | +44 (0)1865 617120 | www.medsci.ox.ac.uk | @OxfordMedSci

On 27 Jul 2017, 10:01 +0100, Andy Haith <notifications@github.com>, wrote:

Thanks Finlay!

**Pull request merged**